samples/apply-style-to-wms-layer/src/main/java/com/esri/arcgismaps/sample/applystyletowmslayer/components/ApplyStyleToWMSLayerViewModel.kt

```diff
@@ -0,0 +1,111 @@
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.WmsLayer
+import com.arcgismaps.mapping.layers.WmsSublayer
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class ApplyStyleToWmsLayerViewModel(app: Application) : AndroidViewModel(app) {
+    // WMS layer displaying Minnesota's county boundaries with multiple styles.
+    private val wmsLayer = WmsLayer(
+        url = "https://imageserver.gisdata.mn.gov/cgi-bin/mncomp?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetCapabilities",
+        layerNames = listOf("mncomp")
+    )
+
+    // Map used by the MapView.
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISLightGray).apply {
+        // apply a min scale
+        minScale = 7_000_000.0
+        // Add the WMS layer to the map's operational layers
+        operationalLayers.add(wmsLayer)
+    }
+
+    // MapViewProxy to perform view operations (like setting the viewpoint) safely from a ViewModel.
+    val mapViewProxy = MapViewProxy()
+
+    // Message dialog ViewModel for error reporting.
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Flow of available WMS styles.
+    private val _styles = MutableStateFlow<List<String>>(emptyList())
+    val styles = _styles.asStateFlow()
+
+    // Selected style index.
+    private val _selectedStyleIndex = MutableStateFlow(0)
+    val selectedStyleIndex = _selectedStyleIndex.asStateFlow()
+
+    // The WMS sublayer for which we will change the style.
+    private var wmsSublayer: WmsSublayer? = null
+
+
+    init {
+        viewModelScope.launch {
+            createAndLoadWmsLayer()
+        }
+    }
+
+    /**
+     * Creates the WMS layer, loads it, zooms to its full extent, and exposes its styles.
+     */
+    private suspend fun createAndLoadWmsLayer() {
+        wmsLayer.load().onSuccess {
+            // Zoom to the full extent of the WMS layer, if available
+            wmsLayer.fullExtent?.let { fullExtent ->
+                mapViewProxy.setViewpointAnimated(Viewpoint(boundingGeometry = fullExtent))
+            }
+
+            // Get the first WMS sublayer
+            wmsSublayer = wmsLayer.sublayers.value.firstOrNull() as? WmsSublayer
+
+            // Obtain available styles from the WMS layer info
+            val styles = wmsLayer.layerInfos.firstOrNull()?.styles ?: emptyList()
+            if (styles.isEmpty()) {
+                messageDialogVM.showMessageDialog("No styles found for the WMS layer.")
+                return@onSuccess
+            }
+
+            _stylesFlow.value = styles
```

Comment:
I'm getting a build error here, request to please ensure the sample is able to run:

```console
sample/applystyletowmslayer/components/ApplyStyleToWMSLayerViewModel.kt:104:22 Unresolved reference '_stylesFlow'.
```

Reply:
apologies, updated.